dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtilityTest.java

```diff
@@ -57,16 +64,25 @@ public void testReadBom_coordinates() throws ArtifactDescriptorException {
   }
 
   @Test
-  public void testReadBom_path() throws MavenRepositoryException {
+  public void testReadBom_path() throws MavenRepositoryException, ArtifactDescriptorException {
     Path pomFile = Paths.get("..", "boms", "cloud-oss-bom", "pom.xml");
     
-    Bom bom = RepositoryUtility.readBom(pomFile);
+    Bom currentBom = RepositoryUtility.readBom(pomFile);
+    Bom oldBom = RepositoryUtility.readBom("com.google.cloud:libraries-bom:2.5.0");
+    ImmutableList<Artifact> currentArtifacts = currentBom.getManagedDependencies();
+    ImmutableList<Artifact> oldArtifacts = oldBom.getManagedDependencies();
+    
+    String coordinates = currentBom.getCoordinates();
+    Truth.assertThat(coordinates).startsWith("com.google.cloud:libraries-bom:");
+    Truth.assertThat(coordinates).endsWith("-SNAPSHOT");
     
-    ImmutableList<Artifact> artifacts = bom.getManagedDependencies();
-    Assert.assertEquals(217, artifacts.size());
-    String coordinates = bom.getCoordinates();
-    Assert.assertTrue(coordinates.startsWith("com.google.cloud:libraries-bom:"));
-    Assert.assertTrue(coordinates.endsWith("-SNAPSHOT"));
+    // This is a characterization test to verify that the managed dependencies haven't changed.
+    // However sometimes this list does change. If so, we want to 
+    // output the specific difference so we can manually verify whether
+    // the changes make sense. When they do make sense, we update the test. 
+    if (currentArtifacts.size() != 217) {
+        Truth.assertThat(currentArtifacts).containsExactly(oldArtifacts);
```

Comment:
```suggestion
        Truth.assertThat(currentArtifacts).containsExactlyElementsIn(oldArtifacts);
```

`containsExactly(Object...)` takes varargs of expected elements. You want `containsExactlyElementsIn(Iterable<?>)`, which takes an iterable of expected elements.

I think Truth actually suggests that in its failure method.

Reply:
And now you've reminded me why I prefer writing out the comparison instead of using yet another library. These method names give me no indication of when I should use `containsExactly` or `containsExactlyElementsIn`. Without looking at the docs I would expect them to do the same thing. I do happen to remember that order is not significant here, but again there's nothing in the method names to tell me that.

If I simply write the code using ordinary Java classes, it's very obvious what is being compared and how and what the conditions are. This change is shorter than the pure-Java equivalent, but it is more opaque. :-(